godtools/App/Features/Dashboard/Domain/UseCases/GetToolFilterCategoriesUseCase/GetToolFilterCategoriesUseCase.swift

```diff
@@ -63,21 +80,28 @@ class GetToolFilterCategoriesUseCase {
                 return nil
             }
             
-            let translatedName: String = localizationServices.stringForLocaleElseEnglish(localeIdentifier: translationLocaleId, key: "tool_category_\(categoryId)")
-            
-            let toolsAvailableText: String = getToolsAvailableText(toolsAvailableCount: toolsAvailableCount, localeId: translationLocaleId)
-            
-            return CategoryFilterDomainModel(
-                type: .category(id: categoryId),
-                translatedName: translatedName,
-                toolsAvailableText: toolsAvailableText,
-                searchableText: translatedName
-            )
+            return self.createCategoryDomainModel(with: categoryId, translationLocaleId: translationLocaleId, filteredByLanguageId: filteredByLanguageId)
         }
         
         return [anyCategory] + categories
     }
     
+    private func createCategoryDomainModel(with categoryId: String, translationLocaleId: String?, filteredByLanguageId: String?) -> CategoryFilterDomainModel {
+        
+        let toolsAvailableCount: Int = getToolsAvailableCount(for: categoryId, filteredByLanguageId: filteredByLanguageId)
+        
+        let translatedName: String = localizationServices.stringForLocaleElseEnglish(localeIdentifier: translationLocaleId, key: "tool_category_\(categoryId)")
+        
+        let toolsAvailableText: String = getToolsAvailableText(toolsAvailableCount: toolsAvailableCount, localeId: translationLocaleId)
+        
+        return CategoryFilterDomainModel(
+            type: .category(id: categoryId),
+            translatedName: translatedName,
+            toolsAvailableText: toolsAvailableText,
+            searchableText: translatedName
+        )
+    }
```

Comment:
I was trying to figure out how to move some of this logic into the `CategoryFilterDomainModel` init, but was having trouble getting there.  Would you think of moving the `getToolsAvailableText()` method into the init instead?  But it can't require `localizationServices`...